src/shorthands/JImage.jl

```diff
@@ -0,0 +1,19 @@
+function _JImage(pos, img, centering, shapeargs, shape, scaleargs)
+    if shape != false
+        shape(shapeargs...)
+    end
+    scale(scaleargs)
```

Comment:
I would really like to have automatic scaling by checking the size of the shape and that of the image. 
For the shape it would be awesome to be able to call it with `:clip` / `:path` automatically but it seems `action` isn't always a keyword argument which makes that a bit hard. Any idea on that @cormullion ?

Reply:
yes, following on from my question (https://github.com/Wikunia/Javis.jl/pull/399#discussion_r689130301) not all graphic construction functions are similar in having an action keyword. We'd have to compile a list and see which ones could be updated...

Reply:
Are you interested in standardizing the functions though such that all have an option which provide using the `action` as a kwarg?

Reply:
Perhaps we can make a list of possible functions as an issue at Luxor.jl. It might be easy or not, not too sure how many are affected...

Reply:
https://github.com/JuliaGraphics/Luxor.jl/issues/174

Reply:
since this spawned off another issue, shall we resolve this comment @Wikunia and @cormullion ?

Reply:
I decided to close this conversation as I do not think automatic scaling is feasible as all shapes can define a bounding box differently within Luxor. 
To find that bounding box for a shape depends on the arguments provided by such a shape.

Reply:
You can find a BoundingBox for a lot of "shapes" using eg `pathtopoly()`:

```julia
@draw begin
    circle(O, 100, :stroke)
    circle(O, 100, :path)
    bbox = BoundingBox(pathtopoly()[1])
    box(bbox, :stroke)
end
```

If you can convert a shape to a polygon, then you can get a bounding box that way. Are there any other shapes in Luxor that you'd like BoundingBox to work on?